programs/compliant_token_template.leo

```diff
@@ -75,25 +102,205 @@ program compliant_token_template.aleo {
+        assert_eq(already_initialized, true);
+        
+        let already_requested: bool = pending_requests.contains(wallet_signing_op_id_hash);
+        if(already_requested) {
+            let existing_pending_op = multisig_core.aleo/pending_signing_ops.get(wallet_signing_op_id_hash);
+            assert(existing_pending_op.expires_at_block < block.height);
+        }
+        assert(!private_pending_requests.contains(wallet_signing_op_id_hash));
+        pending_requests.set(wallet_signing_op_id_hash, multisig_op);
+        init_signing_op_fut.await();
+    }
+
+    // Initializes a new private multisig operation.
+    // Commits the private multisig payload with a salt to compute signing_op_id.
+    // Registers it with the external multisig wallet.
+    // Stores `true` under the hashed wallet operation ID.
+    // Fails if an operation with the same ID already exists unless this request is expired.
+    // Can only be called after the token has been initialized.
+    async transition init_private_multisig_op(wallet_id: address, multisig_op: PrivateCompliantTokenMultisigOp, salt: scalar, public block_expiration: u32) -> (field, field, Future) {
+        let signing_op_id = BHP256::commit_to_field(multisig_op, salt);
+        let (wallet_signing_op_id_hash, init_signing_op_fut) = multisig_core.aleo/initiate_signing_op(wallet_id, signing_op_id, block_expiration);
+        return (signing_op_id, wallet_signing_op_id_hash, fin_init_private_multisig_op(
+            wallet_signing_op_id_hash,
+            init_signing_op_fut
+        ));
+    }
+    async function fin_init_private_multisig_op(
+        wallet_signing_op_id_hash: field,
+        init_signing_op_fut: Future,
+    ) {
+        // Check if the token has already been initialized
+        let already_initialized: bool = token_info.contains(TOKEN_INFO_INDEX);
+        assert_eq(already_initialized, true);
+
+        let already_requested: bool = private_pending_requests.contains(wallet_signing_op_id_hash);
+        if(already_requested) {
+            let existing_pending_op = multisig_core.aleo/pending_signing_ops.get(wallet_signing_op_id_hash);
+            assert(existing_pending_op.expires_at_block < block.height);
+        }
+        assert(!pending_requests.contains(wallet_signing_op_id_hash));
+        private_pending_requests.set(wallet_signing_op_id_hash, true);
+        init_signing_op_fut.await();
+    }
+
     // -------------------------
     // Called by token admins
     // -------------------------
 
+    // Updates the multisig wallet id assigned to a given role.
+    // Generates wallet_signing_op_id_hash using the parameters
+    // Can be executed directly by an address with the manager role,
+    // or by any caller if the corresponding multisig update wallet id request was approved.
+    async transition update_wallet_id(public new_address: address, public role: u16, public multisig_common_params: MultisigCommonParams) -> Future {
```

Comment:
Since this transition actually "Sets role for a multisig wallet", I find the name confusing and suggest renaming to "update_wallet_role"

Reply:
Also, maybe `target_wallet_id` instead of `new_address`?